### ISSUE TITLE
Restarting snapshotter ends up with terminating nydusd

### ISSUE THREAD

**Issue by changweige** [containerd/nydus-snapshotter#106]
```
systemctl restart nydus-snapshotter
```
Then nydusd is terminated ending up with `bash: /bin/ls: Transport endpoint is not connected`

**Comment by jiangliu** [comment]
I can't reproduce this. On my side, it works as expected:
```
$ps -auwx |grep nydus
root      24105  0.0  0.0 197760  3016 pts/1    S    15:09   0:00 sudo /usr/local/bin/containerd-nydus-grpc --address /apsara/gerry/run/containerd-nydus/containerd-nydus-grpc.sock --root /apsara/gerry/var/lib/containerd-nydus-grpc --config-path /apsara/gerry/etc/nydus/nydus-snapshotter.toml --log-level trace --shared-daemon --enable-stargz --nydusd-thread-num 2 --cache-dir /apsara/gerry/var/blobcache
root      24118  0.0  0.0 944596 19800 pts/1    Sl   15:09   0:00 /usr/local/bin/containerd-nydus-grpc --address /apsara/gerry/run/containerd-nydus/containerd-nydus-grpc.sock --root /apsara/gerry/var/lib/containerd-nydus-grpc --config-path /apsara/gerry/etc/nydus/nydus-snapshotter.toml --log-level trace --shared-daemon --enable-stargz --nydusd-thread-num 2 --cache-dir /apsara/gerry/var/blobcache
root      89268 11.2  0.2 9975012 288664 pts/1  Sl   15:08   0:16 /usr/local/bin/nydusd fuse --thread-num 2 --mountpoint /apsara/gerry/var/lib/containerd-nydus-grpc/mnt --apisock /apsara/gerry/var/lib/containerd-nydus-grpc/socket/shared_daemon/api.sock --log-level trace --log-file /apsara/gerry/var/lib/containerd-nydus-grpc/logs/shared_daemon/nydusd.log
liu.jia+ 128115  0.0  0.0 112720   972 pts/1    S+   15:11   0:00 grep --color=auto nydus

[liu.jiang@e18g09468.et15sqa:/home/liu.jiang/scripts]
$sudo kill 24118
[1]+  Done                    sudo /usr/local/bin/containerd-nydus-grpc --address /apsara/gerry/run/containerd-nydus/containerd-nydus-grpc.sock --root /apsara/gerry/var/lib/containerd-nydus-grpc --config-path /apsara/gerry/etc/nydus/nydus-snapshotter.toml --log-level trace --shared-daemon --enable-stargz --nydusd-thread-num 2 --cache-dir /apsara/gerry/var/blobcache

[liu.jiang@e18g09468.et15sqa:/home/liu.jiang/scripts]
$ps -auwx |grep nydus
liu.jia+  13324  0.0  0.0 112716   972 pts/1    S+   15:11   0:00 grep --color=auto nydus
root      89268 10.2  0.2 9975012 288664 pts/1  Sl   15:08   0:16 /usr/local/bin/nydusd fuse --thread-num 2 --mountpoint /apsara/gerry/var/lib/containerd-nydus-grpc/mnt --apisock /apsara/gerry/var/lib/containerd-nydus-grpc/socket/shared_daemon/api.sock --log-level trace --log-file /apsara/gerry/var/lib/containerd-nydus-grpc/logs/shared_daemon/nydusd.log
[liu.jiang@e18g09468.et15sqa:/home/liu.jiang/scripts]
$ps -auwx |grep nydus
liu.jia+  18162  0.0  0.0 112716   972 pts/1    S+   15:11   0:00 grep --color=auto nydus
root      89268 10.0  0.2 9975012 288664 pts/1  Sl   15:08   0:16 /usr/local/bin/nydusd fuse --thread-num 2 --mountpoint /apsara/gerry/var/lib/containerd-nydus-grpc/mnt --apisock /apsara/gerry/var/lib/containerd-nydus-grpc/socket/shared_daemon/api.sock --log-level trace --log-file /apsara/gerry/var/lib/containerd-nydus-grpc/logs/shared_daemon/nydusd.log

[liu.jiang@e18g09468.et15sqa:/home/liu.jiang/scripts]
$sudo /usr/local/bin/containerd-nydus-grpc     --address /apsara/gerry/run/containerd-nydus/containerd-nydus-grpc.sock     --root /apsara/gerry/var/lib/containerd-nydus-grpc     --config-path /apsara/gerry/etc/nydus/nydus-snapshotter.toml     --log-level trace     --shared-daemon     --enable-stargz     --nydusd-thread-num 2     --cache-dir /apsara/gerry/var/blobcache &
[1] 23024

[liu.jiang@e18g09468.et15sqa:/home/liu.jiang/scripts]
$sudo ./hello.sh 
nerdctl --snapshotter nydus run --net none -v /tmp/tmpdt9jhxsy/1:/src -i eci-nydus-registry.cn-hangzhou.cr.aliyuncs.com/v6/java:latest-test_nydus sh
cd /src; javac Main.java; java Main
b'hello\n'
{"repo": "java", "bench": "java:latest-test_nydus", "elapsed": 0.9080977439880371}
```
Logs
```
time="2022-07-28T15:09:46.043819159+08:00" level=info msg="remote mount options [workdir=/apsara/gerry/var/lib/containerd-nydus-grpc/snapshots/13/work upperdir=/apsara/gerry/var/lib/containerd-nydus-grpc/snapshots/13/fs lowerdir=/apsara/gerry/var/lib/containerd-nydus-grpc/mnt/9/fs]"
time="2022-07-28T15:11:26.247205237+08:00" level=info msg="caught signal {}: shutting down"
time="2022-07-28T15:11:26.247381347+08:00" level=info msg="nydus snapshotter exited"
time="2022-07-28T15:11:37.123565525+08:00" level=info msg="gc goroutine start..."
time="2022-07-28T15:11:37.123772470+08:00" level=info msg="found daemon in database" daemon=WCRea_aARrmeKcDpTDnKYw mode=multiple
time="2022-07-28T15:11:37.123793347+08:00" level=info msg="found virtual daemon" daemon=WCRea_aARrmeKcDpTDnKYw
time="2022-07-28T15:11:37.123822120+08:00" level=info msg="found daemon in database" daemon=shared_daemon mode=shared
time="2022-07-28T15:11:37.124466925+08:00" level=info msg="found alive daemon" daemon=shared_daemon
time="2022-07-28T15:11:37.124493640+08:00" level=info msg="found 2 daemons running"
time="2022-07-28T15:11:43.954268796+08:00" level=info msg="view snapshot with key default/15//tmp/initialC2515168513"
time="2022-07-28T15:11:43.954946324+08:00" level=info msg="overlayfs mount options [lowerdir=/apsara/gerry/var/lib/containerd-nydus-grpc/mnt/9/fs:/apsara/gerry/var/lib/containerd-nydus-grpc/snapshots/8/fs:/apsara/gerry/var/lib/containerd-nydus-grpc/snapshots/7/fs:/apsara/gerry/var/lib/containerd-nydus-grpc/snapshots/6/fs:/apsara/gerry/var/lib/containerd-nydus-grpc/snapshots/5/fs:/apsara/gerry/var/lib/containerd-nydus-grpc/snapshots/4/fs:/apsara/gerry/var/lib/containerd-nydus-grpc/snapshots/3/fs:/apsara/gerry/var/lib/containerd-nydus-grpc/snapshots/2/fs:/apsara/gerry/var/lib/containerd-nydus-grpc/snapshots/1/fs]"
time="2022-07-28T15:11:43.980417135+08:00" level=info msg="prepare snapshot with labels map[]" key=default/16/a77a0f75ef8299bbb32c3b00503009e4870859757e0040e33fac5e0ff3152777 parent="default/10/sha256:80d598629ae97c68dda62625a404721f4d3db94ea2d2ec1901bd572e50bbbf40"
time="2022-07-28T15:11:43.980472442+08:00" level=info msg="prepare for container layer default/16/a77a0f75ef8299bbb32c3b00503009e4870859757e0040e33fac5e0ff3152777" key=default/16/a77a0f75ef8299bbb32c3b00503009e4870859757e0040e33fac5e0ff3152777 parent="default/10/sha256:80d598629ae97c68dda62625a404721f4d3db94ea2d2ec1901bd572e50bbbf40"
time="2022-07-28T15:11:43.980502845+08:00" level=info msg="id 15 is data layer, continue to check parent layer"
time="2022-07-28T15:11:43.980577354+08:00" level=info msg="found nydus meta layer id 9, parpare remote snapshot" key=default/16/a77a0f75ef8299bbb32c3b00503009e4870859757e0040e33fac5e0ff3152777 parent="default/10/sha256:80d598629ae97c68dda62625a404721f4d3db94ea2d2ec1901bd572e50bbbf40"
```

**Comment by changweige** [comment]
fi

**Comment by changweige** [comment]
Fixed by #116
